EXOSIMS/Prototypes/ZodiacalLight.py

```diff
@@ -341,80 +361,68 @@ def calcfZmin(self, sInds, Obs, TL, TK, mode, hashname, koMap=None, koTimes=None
+                fZtypes [n, TL.nStars] (~numpy.ndarray(float)):
+                    ndarray of flags for fZmin types that map to fZmins
+                    0 - entering KO
+                    1 - exiting KO
+                    2 - local minimum
+                    max float - not a fZmin candidate
         """
 
         # Generate cache Name
         cachefname = hashname + "fZmin"
 
         # Check if file exists
         if os.path.isfile(cachefname):  # check if file exists
-            self.vprint("Loading cached fZQuads from %s" % cachefname)
-            # load from cache. fZQuads has the form tmpDat len sInds, tmpDat[0] len
-            # number of ko enter/exits and localmin occurences, tmpDat[0,0] form
-            # [type,fZvalue,absTime]
-            with open(cachefname, "rb") as f:  #
-                fZQuads = pickle.load(f)
-                # Convert Abs time to MJD object
-                for i in np.arange(len(fZQuads)):
-                    for j in np.arange(len(fZQuads[i])):
-                        fZQuads[i][j][3] = Time(
-                            fZQuads[i][j][3], format="mjd", scale="tai"
-                        )
-                        fZQuads[i][j][1] = fZQuads[i][j][1] / u.arcsec**2.0
-
-            return [fZQuads[i] for i in sInds]
+            self.vprint("Loading cached fZmins from %s" % cachefname)
+            with open(cachefname, "rb") as f:  # load from cache
+                tmp1 = pickle.load(f, encoding="latin1")
+                fZmins = tmp1["fZmins"]
+                fZtypes = tmp1["fZtypes"]
+            return fZmins, fZtypes
         else:
-            # cast sInds to array
-            sInds = np.array(sInds, ndmin=1, copy=False)
-
-            # this whole block is deprecated
-            # #get all array sizes
-            # nStars = sInds.size
-            # nZ = np.ones(nStars)
-            # valfZmin = nZ * 10 ** (-0.4 * self.magZ) / u.arcsec**2
-            # absTimefZmin = nZ * u.d + TK.currentTimeAbs
+            tmpAssert = np.any(self.fZMap[mode["syst"]["name"]])
+            assert tmpAssert, "fZMap does not exist for the mode of interest"
 
             tmpfZ = np.asarray(self.fZMap[mode["syst"]["name"]])
-            fZ_matrix = tmpfZ[sInds, :]  # Apply previous filters to fZMap[sInds, 1000]
-            dt = 365.25 / len(np.arange(1000))
-            timeArray = [j * dt for j in np.arange(1000)]
-            timeArrayAbs = TK.currentTimeAbs + timeArray * u.d
+            fZ_matrix = tmpfZ[sInds, :]  # Apply previous filters to fZMap
 
             # When are stars in KO regions
-            missionLife = TK.missionLife.to("yr")
-            # if this is being calculated without a koMap,
-            # or if missionLife is less than a year
-            if (koMap is None) or (missionLife.value < 1):
+            # if calculated without a koMap
+            if koMap is None:
+                assert koTimes is None, "Corresponding koMap not included."

```

Comment:
Why is this assert needed here? You're just overwriting koTimes in the next line, so why throw an error if it exists?  I also don't think the error message is particularly informative.  This might be a warning, although I'm not actually sure how you'd get a koTimes without a map?